elasticapm/instrumentation/packages/base.py

```diff
@@ -219,7 +219,8 @@ def uninstrument(self):
         self.originals = {}
 
     def call_if_sampling(self, module, method, wrapped, instance, args, kwargs):
-        if not get_transaction():
+        transaction = get_transaction()
+        if not transaction or not transaction.is_sampled:
             return wrapped(*args, **kwargs)
```

Comment:
you have the check in `begin_span`. Do you need it here too?

Reply:
not 100% sure tbh. I'll step through it with the debugger

Reply:
After taking a closer read of the code, I think it's worth doing the check in both places.

`call_if_sampling` is called for stuff we instrument automatically. Checking here avoids even calling the `capture_span` decorator and doing other stuff we do in the wrappers (e.g. parsing the SQL query for an `sql` span).

the check in `capture_span.__enter__` is useful for cases where code is instrumented manually using the `capture_span` decorator.

The third check, in `Transaction.begin_span` seems to be superfluous, however, as it is only called from `capture_span.__enter__`. Good catch!